### ISSUE TITLE
Issues with verify/update generated docs on OS X

### ISSUE THREAD

**Issue by a-robinson** [kubernetes/kubernetes#22317]
Running hack/update-generated-docs.sh on my macbook pro at OS X version 10.11.2 spits out the following error:

``` shell
$ ./hack/update-generated-docs.sh
+++ [0301 15:35:38] Building go targets for darwin/amd64:
    cmd/gendocs
    cmd/genkubedocs
    cmd/genman
    cmd/genbashcomp
    cmd/mungedocs
+++ [0301 15:35:40] Placing binaries
/Users/arob/kubernetes/hack/lib/util.sh: line 234: mdfiles[@]: unbound variable
```

Running hack/verify-generated-docs.sh hits the same error indirectly:

``` shell
$ ./hack/verify-generated-docs.sh
+++ [0301 15:36:17] Building go targets for darwin/amd64:
    cmd/gendocs
    cmd/genkubedocs
    cmd/genman
    cmd/genbashcomp
    cmd/mungedocs
+++ [0301 15:36:19] Placing binaries
/Users/arob/kubernetes/hack/lib/util.sh: line 234: mdfiles[@]: unbound variable
!!! Error in /Users/arob/kubernetes/hack/after-build/verify-generated-docs.sh:68
  'needsanalytics=($(kube::util::gen-analytics "${KUBE_ROOT}" 1))' exited with status 1
Call stack:
  1: /Users/arob/kubernetes/hack/after-build/verify-generated-docs.sh:68 main(...)
Exiting with status 1
!!! Error in ./hack/verify-generated-docs.sh:28
  '"${KUBE_ROOT}/hack/after-build/verify-generated-docs.sh" "$@"' exited with status 1
Call stack:
  1: ./hack/verify-generated-docs.sh:28 main(...)
Exiting with status 1
```

It may be due to differing version of `find` than are on linux. For what it's worth, my docker version:

``` shell
$ docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      darwin/amd64

Server:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        2016-02-22T22:37:33.778002647+00:00
 OS/Arch:      linux/amd64
```

**Comment by imkin** [comment]
I ran ./hack/update-generated-docs.sh and see this error. 

I am on ubuntu -wily

 ./hack/update-generated-docs.sh 
+++ [0302 10:54:45] Building go targets for linux/amd64:
    cmd/gendocs
    cmd/genkubedocs
    cmd/genman
    cmd/genbashcomp
    cmd/mungedocs
# k8s.io/kubernetes/cmd/gendocs
_output/local/go/src/k8s.io/kubernetes/cmd/gendocs/gen_kubectl_docs.go:51: undefined: cobra.GenMarkdownTree
df# k8s.io/kubernetes/cmd/genkubedocs
_output/local/go/src/k8s.io/kubernetes/cmd/genkubedocs/gen_kube_docs.go:54: undefined: cobra.GenMarkdownTree
_output/local/go/src/k8s.io/kubernetes/cmd/genkubedocs/gen_kube_docs.go:58: undefined: cobra.GenMarkdownTree
_output/local/go/src/k8s.io/kubernetes/cmd/genkubedocs/gen_kube_docs.go:62: undefined: cobra.GenMarkdownTree
_output/local/go/src/k8s.io/kubernetes/cmd/genkubedocs/gen_kube_docs.go:66: undefined: cobra.GenMarkdownTree
_output/local/go/src/k8s.io/kubernetes/cmd/genkubedocs/gen_kube_docs.go:70: undefined: cobra.GenMarkdownTree

!!! Error in /playground/projects/vmware/kubernetes/hack/lib/golang.sh:433
  'go install "${goflags[@]:+${goflags[@]}}" -ldflags "${goldflags}" "${nonstatics[@]:+${nonstatics[@]}}"' exited with status 2
Call stack:
  1: /playground/projects/vmware/kubernetes/hack/lib/golang.sh:433 kube::golang::build_binaries_for_platform(...)
  2: /playground/projects/vmware/kubernetes/hack/lib/golang.sh:575 kube::golang::build_binaries(...)
  3: /playground/projects/vmware/kubernetes/hack/build-go.sh:26 main(...)
Exiting with status 1
!!! Error in /playground/projects/vmware/kubernetes/hack/lib/golang.sh:496
  '( kube::golang::setup_env; local host_platform; host_platform=$(kube::golang::host_platform); local goflags goldflags; eval "goflags=(${KUBE_GOFLAGS:-})"; goldflags="${KUBE_GOLDFLAGS:-} $(kube::version::ldflags)"; local use_go_build; local -a targets=(); local arg; for arg in "$@";
do
    if [[ "${arg}" == "--use_go_build" ]]; then
        use_go_build=true;
    else
        if [[ "${arg}" == -* ]]; then
            goflags+=("${arg}");
        else
            targets+=("${arg}");
        fi;
    fi;
done; if [[ ${#targets[@]} -eq 0 ]]; then
    targets=("${KUBE_ALL_TARGETS[@]}");
fi; local -a platforms=("${KUBE_BUILD_PLATFORMS[@]:+${KUBE_BUILD_PLATFORMS[@]}}"); if [[ ${#platforms[@]} -eq 0 ]]; then
    platforms=("${host_platform}");
fi; local binaries; binaries=($(kube::golang::binaries_from_targets "${targets[@]}")); local parallel=false; if [[ ${#platforms[@]} -gt 1 ]]; then
    local gigs; gigs=$(kube::golang::get_physmem); if [[ ${gigs} -ge ${KUBE_PARALLEL_BUILD_MEMORY} ]]; then
        kube::log::status "Multiple platforms requested and available ${gigs}G >= threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in parallel"; parallel=true;
    else
        kube::log::status "Multiple platforms requested, but available ${gigs}G < threshold ${KUBE_PARALLEL_BUILD_MEMORY}G, building platforms in serial"; parallel=false;
    fi;
fi; if [[ "${parallel}" == "true" ]]; then
    kube::log::status "Building go targets for ${platforms[@]} in parallel (output will appear in a burst when complete):" "${targets[@]}"; local platform; for platform in "${platforms[@]}";
    do
        ( kube::golang::set_platform_envs "${platform}"; kube::log::status "${platform}: go build started"; kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-}; kube::log::status "${platform}: go build finished" ) &> "/tmp//${platform//\//_}.build" &
    done; local fails=0; for job in $(jobs -p);
    do
        wait ${job} || let "fails+=1";
    done; for platform in "${platforms[@]}";
    do
        cat "/tmp//${platform//\//_}.build";
    done; exit ${fails};
else
    for platform in "${platforms[@]}";
    do
        kube::log::status "Building go targets for ${platform}:" "${targets[@]}"; kube::golang::set_platform_envs "${platform}"; kube::golang::build_binaries_for_platform ${platform} ${use_go_build:-};
    done;
fi )' exited with status 1
Call stack:
  1: /playground/projects/vmware/kubernetes/hack/lib/golang.sh:496 kube::golang::build_binaries(...)
  2: /playground/projects/vmware/kubernetes/hack/build-go.sh:26 main(...)
Exiting with status 1
!!! Error in ./hack/update-generated-docs.sh:26
  '"${KUBE_ROOT}/hack/build-go.sh" cmd/gendocs cmd/genkubedocs cmd/genman cmd/genbashcomp cmd/mungedocs' exited with status 1
Call stack:
  1: ./hack/update-generated-docs.sh:26 main(...)
Exiting with status 1



╭─superman@localhost /playground/projects/vmware/kubernetes git ➦ (vsphere-pod-cidr-fix) 2016-03-02 10:56:26 Wednesday
╰─$ docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:20:08 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:20:08 UTC 2015
 OS/Arch:      linux/amd64

**Comment by vishh** [comment]
@bgrant0607: Do you know who maintains the docs scripts?

**Comment by bgrant0607** [comment]
cc @eparis @caesarxuchao @nikhiljindal @pwittrock @johndmulhausen

**Comment by bgrant0607** [comment]
If we want Mac OS to work, we need CI for it.

**Comment by pwittrock** [comment]
Moving the scripts to run in docker containers may be the way to provide cross platform support (albeit through a vm under the covers)

**Comment by bgrant0607** [comment]
There's another issue for eliminating generated docs, which is what we should do
